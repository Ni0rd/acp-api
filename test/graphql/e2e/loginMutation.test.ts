import './env';
import gql from 'graphql-tag';
import { addGlobalMocks, initApollo } from './helpers';

addGlobalMocks();
const getClient = initApollo();

jest.mock('../../../api/utils/auth', () => {
  return {
    login: jest.fn().mockResolvedValue({
      id: 1,
      email: 'email@example.com',
      firstname: 'John',
      lastname: 'Doe',
    }),
    signToken: jest.fn().mockResolvedValue('abcdefg'),
    getDecodedTokenFromHeaders: jest.fn().mockReturnValue(null),
  };
});

describe('login mutation', () => {
  test('returns a LoginResult', async () => {
    const mutation = {
      mutation: gql`
        mutation($email: EmailAddress!, $password: String!) {
          login(email: $email, password: $password) {
            token
            me {
              id
              email
              firstname
              lastname
            }
          }
        }
      `,
      variables: {
        email: 'email@example.com',
        password: '123456',
      },
    };
    const response = await getClient().mutate(mutation);
    expect(response).toMatchSnapshot();
  });
});
