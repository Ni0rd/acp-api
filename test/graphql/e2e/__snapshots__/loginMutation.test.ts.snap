// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login mutation returns a LoginResult 1`] = `
Object {
  "data": Object {
    "login": Object {
      "__typename": "LoginResult",
      "me": Object {
        "__typename": "User",
        "email": "email@example.com",
        "firstname": "John",
        "id": 1,
        "lastname": "Doe",
      },
      "token": "abcdefg",
    },
  },
}
`;
